File: src/c/_2038_Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color.c
bool winnerOfGame(char *colors)
{
    int moveA = 0, moveB = 0, countA = 0, countB = 0;

    while (*colors)
    {
        if (*colors == 'A')
        {
            countB = 0;
            countA++;
            if (countA > 2)
                moveA++;
        }
        else
        {
            countA = 0;
            countB++;
            if (countB > 2)
                moveB++;
        }
        colors++;
    }

    return moveA > moveB;
}

File: src/python/_2038_Remove_Colored_Pieces_if_Both_Neighbors_are_the_Same_Color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        moveA, moveB, countA, countB = 0, 0, 0, 0

        for color in colors:
            if color == "A":
                countB = 0
                countA += 1
                if countA > 2:
                    moveA += 1
            else:
                countA = 0
                countB += 1
                if countB > 2:
                    moveB += 1

        return moveA > moveB
